### PR TITLE
CT-1782 add tests for deleting an ICO case

### PR DESCRIPTION
**Pull request by beccapearce**
This tests deleting an open ICO we should add a test for closed
ICOs once they're available following the same pattern as the test for
deleting a SAR case

### CODE REVIEW

**Review by aliuk2012 — COMMENTED** [review]
Just for consistency, could `spec/features/cases/ico/deleting_ico_spec.rb` be renames to case_deleting_spec.rb like SARS or perhaps updates SAR.

File: spec/controllers/cases_controller/destroy_spec.rb [hunk 1]
```diff
@@ -8,22 +8,35 @@
...
+        unassigned_case.reload
+        expect(unassigned_case.deleted?).to be true
+      end
     end
 
-    it 'marks the case as deleted' do
-      delete :destroy, params: params
-      unassigned_case.reload
-      expect(unassigned_case.deleted?).to be true
+    context 'ICO case' do
```
Comment (aliuk2012) [inline spec/controllers/cases_controller/destroy_spec.rb]: Should there also be a SAR version of this test?

Comment (beccapearce) [reply spec/controllers/cases_controller/destroy_spec.rb]: As in a normal SAR case or an ICO SAR case?
I thought it about a normal SAR case, but this ticket was about ICOs, I can add one in though.
ICO SAR cases I think are similar enough to ICO FOIs, but equally not too much work to add it in so can do

Comment (aliuk2012) [reply spec/controllers/cases_controller/destroy_spec.rb]: Sorry, I forgot we had ICO SAR, I was referring to standard SAR cases.

Comment (beccapearce) [reply spec/controllers/cases_controller/destroy_spec.rb]: I'll do both I think, It's so quick might as well!

**Review by aliuk2012 — APPROVED** [review]

**Pull request merged**